src/extlibs/hrvo/hrvo_agent.h

```diff
@@ -153,6 +153,60 @@ class HRVOAgent : public Agent
     // velocity
     static constexpr float MIN_PREF_SPEED_MULTIPLER = 0.4f;
 
+    /**
+     * Returns the first velocity obstacle intersected by the Candidate point in
+     * velocityObstacles_
+     *
+     * @param candidate the candidate point to check against all velocity obstacles in
+     * velocityObstacles_
+     *
+     * @return the first velocity obstacle that the given candidate point intersects, or
+     * -1 if it doesn't intersect anything
+     */
+    int findIntersectingVelocityObstacle(const Candidate &candidate) const;
```

Comment:
we should just do std optional instead of -1

Reply:
done